src/components/PatentVisualizer/PatentVisualizer.js

```diff
@@ -1,193 +1,207 @@
-                    toggleBaseline={toggleBaseline}
-                />
-                <Layout style={{ padding: '24px' }}>
-                    <Heatmap onEvent={onEvent} {...config} />
-                </Layout>
-                {details.show && 
-                <Sider className="site-layout-background" theme="light" width={200} style={{ padding: '20px' }}>
-                    <CloseOutlined className="visualizer__details-sider-icon" 
-                        onClick={() => setDetails({ ...details, show: false })}
-                    />
-                    <div>Patent Name: {details.patentId}</div>
-                    <div>Sequence Index - Name: {details.seqPosition}</div>
-                    <div>Assignee: {details.assignee}</div>
-                    <div>Other relevant data: </div>
-                    <div>Link to PDF</div>
-                    <div>Text extracted</div>
-                    <div className="visualizer__edit-button-container">
-                        <Button type="primary">Edit Data</Button>
-                    </div>
-                </Sider>
-                }
-            </Layout>,
-            <PatentTable />
-        ]
-    )
-};
-
+import React, { useState, useEffect, useRef } from 'react';
+import 'antd/dist/antd.css';
+import './PatentVisualizer.css';
+import { CloseOutlined } from '@ant-design/icons';
+import { Heatmap } from '@ant-design/charts';
+import { Layout, Button } from 'antd';
+import PatentVisualizerSidebar from './PatentVisualizerSidebar';
+import PatentTable from './PatentTable';
+import { assignColors } from '../../utils/colors';
+import { getUnique } from '../../utils/utils';
+import { getPatentData, generateVisualizationDataset } from '../../utils/patentDataUtils';
+import { useLocation } from 'react-router-dom';
+
+const { Sider } = Layout;
+
+const KEYS = {
+    assignee: 'patentAssignees',
+    sequencePosition: 'sequencePosition',
+    patentNumber: 'patentNumber',
+    claimed: 'Claimed',
+    aminoAcid: 'Amino Acid',
+    baseline: 'Sequence'
+}
+Object.freeze(KEYS);
+
+const getMaximumSeq = (patentArray) => {
+    return patentArray.reduce((max, current) => {
+        const seqPosition = parseInt(current[KEYS.sequencePosition]);
+        if(seqPosition > max) {
+            return seqPosition;
+        } else {
+            return max;
+        }
+    }, 0);
+}
+
+const PatentVisualizer = props => {
+    const location = useLocation();
+
+    const [data, setData] = useState([]);
+    const [colorKeys, setColorKeys] = useState({});
+    const [details, setDetails] = useState({ show: false, patentId: 0, seqPosition: 0 });
+    const [showBaseline, setBaseline] = useState(false);
+    const _dataRef = useRef([]);
+
+    useEffect(() => {
+        const proteinName = location.state.proteinName;
+        console.log('Fetching patent details for', proteinName)
+        const patentData = getPatentData(proteinName);

```

Comment:
in my mind this would be an async call and return a promise (clearly not). How is it that `API.get` makes the value directly accessible

Reply:
is https://github.com/patent-analysis/psv/pull/73/commits/fae3cff6fffd436708d517bbf96d2d920c5de392 what you were looking for?

Reply:
this was just a question, as it looked like API.get was already returning the response from the API directly instead of wrapping it as a promise. We do not need https://github.com/patent-analysis/psv/commit/fae3cff6fffd436708d517bbf96d2d920c5de392 if it was already working.

Reply:
okay, do you want me to revert this change or is this PR good to go?

Reply:
if we dont need it id revert it